app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/setup/OAuthAppHandler.java

```diff
@@ -139,6 +140,16 @@ public void update(@NotNull @PathParam("id") String id, @NotNull @Valid OAuthApp
             .build();
 
         dataMgr.update(updated);
+
+        final boolean shouldBeDerived = app.clientId != null && app.clientSecret != null;
+
+        dataMgr.fetchAllByPropertyValue(Connection.class, "connectorId", id).forEach(connection -> toggleDerived(connection, shouldBeDerived));
```

Comment:
Is the change on the stream really persisted ?

Reply:
A bit further down in the `toggleDerived` method:

https://github.com/syndesisio/syndesis/pull/2534/files#diff-ed861ccdc3b0bd650c4032e9e633654cR152